springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/AsyncPublisher.java

```diff
@@ -0,0 +1,21 @@
+package io.github.stavshamir.springwolf.asyncapi.scanners.channels.operationdata.annotation;
+
+import io.github.stavshamir.springwolf.asyncapi.types.OperationData;
+import io.github.stavshamir.springwolf.asyncapi.types.ProducerData;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation is mapped to {@link ProducerData}
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(value = {ElementType.METHOD})
+public @interface AsyncPublisher {
```

Comment:
What I'm missing here, when I'm taking the user POV, is where can I use this? Can every method be annotated by this? If not, what is the signature structure/pattern that is allowed? 
It looks like the payload type can be extracted if the method has a certain signature. If it doesn't, the javadoc should explain how to specify the payload type explicitly, etc.